tests/Packetizer/CCSDSPacketizerTests.cc

```diff
@@ -0,0 +1,231 @@
+#include "CppUTest/TestHarness.h"
+
+#include <cstring>
+
+extern "C"
+{
+#include "IsoChecksum.h"
+#include "CCSDSPacketizer.h"
+#include "PacketizerInternal.h"
+#include "SpacePacketInternal.h"
+}
+
+TEST_GROUP(CCSDSPacketizer)
+{
+    static constexpr size_t dataSize = 514;
+    static constexpr size_t packetSize = SPACE_PACKET_PRIMARY_HEADER_SIZE + dataSize + SPACE_PACKET_ERROR_CONTROL_SIZE;
+    uint8_t packetData[packetSize];
+
+    void setup() { memset(packetData, 0, packetSize); }
+};
+
+TEST(CCSDSPacketizer, CCSDSPacketizeTelemetry)
+{
+    const uint16_t destination = 682;
+    const uint16_t sequenceCount = 792;
+
+    Packetizer packetizer{ sequenceCount };
+
+    auto resultSize = CCSDS_Packetizer_packetize(&packetizer,
+                                                 Packetizer_PacketType_Telemetry,
+                                                 0,
+                                                 destination,
+                                                 packetData,
+                                                 SPACE_PACKET_PRIMARY_HEADER_SIZE,
+                                                 dataSize);
+
+    CHECK_EQUAL(packetSize, resultSize);
+    CHECK_EQUAL(0b00001010, packetData[0]);
```

Comment:
These tests are nice, because they have a "fixed" reference. However, they are hard to understand by a human. Would it be possible to add at least one test, which first packetizes and then depacketizes a packet, and checks whether the output data and meta is equal to the one on the input?